### PR TITLE
Strict petsc error code compatibility

### PR DESCRIPTION
**Pull request by lindsayad**
Some PETSc devs only configure with `--with-strict-petscerrorcode` and if I'd ever like them to look at something, it's a lot easier if our stack builds out of the box for them

### CODE REVIEW

**Review by roystgnr — APPROVED** [review]
I thought maybe the m4 changes were more `CXXFLAGS=-Werror configure` stuff I missed, but that wouldn't explain having to explicitly cast from int to PetscErrorCode.  What PETSc/compiler version was complaining about the implicit casts there?

**Comment by lindsayad** [comment]
You get these errors/warnings when you configure PETSc `--with-strict-petscerrorcode`

File: include/numerics/petsc_vector.h [hunk 1]
```diff
@@ -1187,7 +1187,7 @@ Real PetscVector<T>::min () const
 
   this->_restore_array();
 
-  PetscErrorCode ierr=0;
+  PetscErrorCode ierr = (PetscErrorCode)0;
```
Comment (jwpeterson) [inline include/numerics/petsc_vector.h:1190]: @lindsayad does PETSc `--with-strict-petscerrorcode` warn/error about about C++ style casts, e.g.
```
auto ierr = static_cast<PetscErrorCode>(0);
```
It is a small thing but I prefer to use C++ style casts whenever possible.

Comment (lindsayad) [reply include/numerics/petsc_vector.h:1190]: It works fine. I've opened https://github.com/libMesh/libmesh/pull/3867 to change it

Comment (jwpeterson) [reply include/numerics/petsc_vector.h:1190]: OK, thanks, I was going to just update it at some later point, didn't mean to imply that you should do it :)

**Pull request merged**